chore: qualify screen keywords

J-Son89 opened the pull request:
This pr brings no functional changes. 
All it does is add some syntactic benefits to code readability and maintenance.

These adjustments will only affect (if anywhere) the wallet screens, it's best to do a quick smoke test of the wallet and the various screens there.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 88aaff79 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-18778/1/) | 2024-02-09 15:42:42 | ~6 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/ccrsk9) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2Fccrsk9)|

status-im-auto commented:
#### :heavy_check_mark: [status-mobile/prs/android/PR-18778#1](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-18778/1/) :small_blue_diamond: ~9 min 1 sec :small_blue_diamond: 88aaff79 :small_blue_diamond: [:package: android package](https://status-im-mobile-prs.ams3.digitaloceanspaces.com/StatusIm-Mobile-240209-153638-88aaff-pr18778-universal.apk)

J-Son89 commented on src/status_im/common/universal_links.cljs:
```diff
@@ -98,7 +98,7 @@
   [cofx {address :account}]
   (when-let [account (existing-account? cofx address)]
     (navigation/navigate-to cofx
-                            :wallet-account
```
I think this actually goes to a dead end now as is. cc @yqrashawn - regardless of this pr going in we probably should point this to the new wallet screens 👍

J-Son89 commented on src/status_im/navigation/view.cljs:
```diff
@@ -52,12 +52,18 @@
-                                                (get-screens)
-                                                screens)
-                                              (keyword screen-key))
+     (let [screen-details (get (if js/goog.DEBUG
```
a pretty fugly solution to resolve the screens options etc.

If we take this approach this would be temporary as we eventually would have all screens using  `(keyword "screen" screen-key)` 👍

mohsen-ghafouri commented:
Appreciate you raising this point and for your work in improving the flow.

While I see the intention behind adding the `:screen` keyword for clarity, I'm wondering if it might be a bit redundant, considering the context of navigation. Typically, when we navigate, it's already implied that we're moving to a screen. 

However, I'm open to exploring scenarios where this extra qualification could be helpful or prevent any confusion

J-Son89 commented:
> Appreciate you raising this point and for your work in improving the flow.
> 
> While I see the intention behind adding the `:screen` keyword for clarity, I'm wondering if it might be a bit redundant, considering the context of navigation. Typically, when we navigate, it's already implied that we're moving to a screen.
> 
> However, I'm open to exploring scenarios where this extra qualification could be helpful or prevent any confusion

yeah when using a keyword we are pretty much always aware of the use case and so that is not what I am trying to improve here.
The most valuable reason I see for this is that when doing later refactors of screen keywords (or any other keywords) that we can safely update the keywords without affecting other unintended domains/ aspects of the codebase. Without qualification we have to look quite precisely at what each keywords purpose is and that is both tedious and error prone.

There might be further use cases (or drawbacks) @ulisesmac, @ilmotta or our other more experienced Clojurists might 
be aware of.

shivekkhurana commented:
> The most valuable reason I see for this is that when doing later refactors of screen keywords (or any other keywords) that we can safely update the keywords without affecting other unintended domains/ aspects of the codebase. Without qualification we have to look quite precisely at what each keywords purpose is and that is both tedious and error prone.

I don't understand this. If we want a refactor related to screens, we can easily look up for contexts, or do a project with search for `navigate`. 

Not able to understand what value this brings (maybe I'm wrong re: refactoring? please correct me if that's the case).

I do see a downside. Having arbitrary keywords will pollute the codebase. I know with `:screen`, your intent is very clear. But there could be a case where navigation opens a `bottom-sheet`. It will be very enticing to have a `:bottom-sheet` keyword in that case. There is no end to this.

ulisesmac commented:
> > Appreciate you raising this point and for your work in improving the flow.
> > While I see the intention behind adding the `:screen` keyword for clarity, I'm wondering if it might be a bit redundant, considering the context of navigation. Typically, when we navigate, it's already implied that we're moving to a screen.
> > However, I'm open to exploring scenarios where this extra qualification could be helpful or prevent any confusion
> 
> yeah when using a keyword we are pretty much always aware of the use case and so that is not what I am trying to improve here. (well in my experience so far) The most valuable reason I see for this is that when doing later refactors of screen keywords (or any other keywords) that we can safely update the keywords without affecting other unintended domains/ aspects of the codebase. Without qualification we have to look quite precisely at what each keywords purpose is and that is both tedious and error prone.
> 
> There might be further use cases (or drawbacks) @ulisesmac, @ilmotta or our other more experienced Clojurists might be aware of.

@mohsen-ghafouri 
Some other aspects improved are the  code organization  and the dev experience while writing keywords. It's less error-prone to write because the editor suggest the keys:

Currently with icons:
![image](https://github.com/status-im/status-mobile/assets/90291778/909ca8e0-0595-4260-8f6c-571b2cf12844)

Currently with translations:
![image](https://github.com/status-im/status-mobile/assets/90291778/e1d1b14d-48d3-4772-8cb1-46ff28f19966)

briansztamfater commented:
> > > Appreciate you raising this point and for your work in improving the flow.
> > > While I see the intention behind adding the `:screen` keyword for clarity, I'm wondering if it might be a bit redundant, considering the context of navigation. Typically, when we navigate, it's already implied that we're moving to a screen.
> > > However, I'm open to exploring scenarios where this extra qualification could be helpful or prevent any confusion
> > 
> > 
> > yeah when using a keyword we are pretty much always aware of the use case and so that is not what I am trying to improve here. (well in my experience so far) The most valuable reason I see for this is that when doing later refactors of screen keywords (or any other keywords) that we can safely update the keywords without affecting other unintended domains/ aspects of the codebase. Without qualification we have to look quite precisely at what each keywords purpose is and that is both tedious and error prone.
> > There might be further use cases (or drawbacks) @ulisesmac, @ilmotta or our other more experienced Clojurists might be aware of.
> 
> @mohsen-ghafouri Some other aspects improved are the code organization and the dev experience while writing keywords. It's less error-prone to write because the editor suggest the keys:
> 
> Currently with icons: ![image](https://private-user-images.githubusercontent.com/90291778/304182761-909ca8e0-0595-4260-8f6c-571b2cf12844.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5MzE1MjYsIm5iZiI6MTcwNzkzMTIyNiwicGF0aCI6Ii85MDI5MTc3OC8zMDQxODI3NjEtOTA5Y2E4ZTAtMDU5NS00MjYwLThmNmMtNTcxYjJjZjEyODQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDE3MjAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkYjM4MjFlYWZjNWQ0NTA1MjQyYzhkNTc4YWExNjkxOTU4NzMxOGUzZWRiMWRmMTNhNTYyZWJmODYwZmViZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tN21BZP0vQwPBz6vh1ABNrQzNB7bLjASgUigWM6jDAk)
> 
> Currently with translations: ![image](https://private-user-images.githubusercontent.com/90291778/304182833-e1d1b14d-48d3-4772-8cb1-46ff28f19966.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5MzE1MjYsIm5iZiI6MTcwNzkzMTIyNiwicGF0aCI6Ii85MDI5MTc3OC8zMDQxODI4MzMtZTFkMWIxNGQtNDhkMy00NzcyLThjYjEtNDZmZjI4ZjE5OTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDE3MjAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3YmEyODZlMTQ4OTVhMTNkMGRjNmQxNjY4OTk2MDRiMWViMjY1MTAxNmM1YTViMmM1MDRjNTExNTQ5YzUyZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yF9myWs6nEPFkAbQh_gGFAJRl13xSfQW87aaAUO4pRQ)

If we use `:i/` for icons and `:t/` for translations, maybe use `:s/` instead of `:screen/`?

Parveshdhull commented on src/status_im/navigation/view.cljs:
```diff
@@ -52,12 +52,18 @@
                                                 (get-screens)
                                                 screens)
                                               (keyword screen-key))
+           qualified-screen-details      (get (if js/goog.DEBUG
+                                                (get-screens)
+                                                screens)
+                                              (keyword "screen" screen-key))
```
Hi @J-Son89, Please can you elaborate why this is needed?

J-Son89 replied:
For sure, 
As far as I understand it,  when we navigate we use the `name` function to pass the screen name to the navigation package. `name` removes the qualification and so we have to add the `:screen/` qualification back here. 
It's not such a change as in either case we need this `keyword` processing here anyway but we need to support both until we would have completely migrated `keywords` over.

J-Son89 replied:
there could be some other approaches to handling this too.

Parveshdhull replied:
the name doesn't remove this qualification, this was done by the `(clj->js)` function. It seems that these keyword qualifications
 are not playing well with JavaScript.

I think we should be a little careful to use them until we figure out a solution for this. Because we are covering this specific case, we might miss some other places and cause unintentional bugs.

J-Son89 replied:
perhaps not in this case, but it does:
https://cljs.github.io/api/cljs.core/name
<img width="212" alt="Screenshot 2024-02-15 at 13 29 49" src="https://github.com/status-im/status-mobile/assets/22799766/2650c10d-27ff-4acd-8f25-51d07e43ce16">

Not so sure, from what I seen this is the only place this happens. Will investigate more as it's a worthy concern and maybe it can be handled cleaner. In this particular case we still have to be careful to modify the keywords from a string anyway.

Parveshdhull replied:
Oh sorry, I thought you meant `:name` parameter is getting stripped inside the navigation package (as that's what we are using while navigating).

Anyway, thank you for pointing out that the `name` function also has `clj->js` limitations while working with js.

J-Son89 replied:
Ah okay, gotcha. In any case I'll do a further check to ensure there's no too many vulnerabilities related to this.

ulisesmac replied:
If this works for something, we can preserve the namespace by extracting `fqn` from a keyword. E.g.

`(.-fqn :my/keyword)`
;; => `"my/keyword"`

J-Son89 commented:
> > > > Appreciate you raising this point and for your work in improving the flow.
> > > > While I see the intention behind adding the `:screen` keyword for clarity, I'm wondering if it might be a bit redundant, considering the context of navigation. Typically, when we navigate, it's already implied that we're moving to a screen.
> > > > However, I'm open to exploring scenarios where this extra qualification could be helpful or prevent any confusion
> > > 
> > > 
> > > yeah when using a keyword we are pretty much always aware of the use case and so that is not what I am trying to improve here. (well in my experience so far) The most valuable reason I see for this is that when doing later refactors of screen keywords (or any other keywords) that we can safely update the keywords without affecting other unintended domains/ aspects of the codebase. Without qualification we have to look quite precisely at what each keywords purpose is and that is both tedious and error prone.
> > > There might be further use cases (or drawbacks) @ulisesmac, @ilmotta or our other more experienced Clojurists might be aware of.
> > 
> > 
> > @mohsen-ghafouri Some other aspects improved are the code organization and the dev experience while writing keywords. It's less error-prone to write because the editor suggest the keys:
> > Currently with icons: ![image](https://private-user-images.githubusercontent.com/90291778/304182761-909ca8e0-0595-4260-8f6c-571b2cf12844.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5MzE1MjYsIm5iZiI6MTcwNzkzMTIyNiwicGF0aCI6Ii85MDI5MTc3OC8zMDQxODI3NjEtOTA5Y2E4ZTAtMDU5NS00MjYwLThmNmMtNTcxYjJjZjEyODQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDE3MjAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkYjM4MjFlYWZjNWQ0NTA1MjQyYzhkNTc4YWExNjkxOTU4NzMxOGUzZWRiMWRmMTNhNTYyZWJmODYwZmViZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tN21BZP0vQwPBz6vh1ABNrQzNB7bLjASgUigWM6jDAk)
> > Currently with translations: ![image](https://private-user-images.githubusercontent.com/90291778/304182833-e1d1b14d-48d3-4772-8cb1-46ff28f19966.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5MzE1MjYsIm5iZiI6MTcwNzkzMTIyNiwicGF0aCI6Ii85MDI5MTc3OC8zMDQxODI4MzMtZTFkMWIxNGQtNDhkMy00NzcyLThjYjEtNDZmZjI4ZjE5OTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDE3MjAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3YmEyODZlMTQ4OTVhMTNkMGRjNmQxNjY4OTk2MDRiMWViMjY1MTAxNmM1YTViMmM1MDRjNTExNTQ5YzUyZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yF9myWs6nEPFkAbQh_gGFAJRl13xSfQW87aaAUO4pRQ)
> 
> If we use `:i/` for icons and `:t/` for translations, maybe use `:s/` instead of `:screen/`?

We do have some other keywords that are more verbose too.
for example `:social/`, `:token/`.
I think the decision we were basing it off for when we want to use the abbreviated form vs more verbose is how frequently we use these. right @ilmotta? 🤔 
and I guess something else to consider is there any other qualified keyword that would benefit more from having `:s/` instead? 🤔

J-Son89 commented:
> I don't understand this. If we want a refactor related to screens, we can easily look up for contexts, or do a project with search for `navigate`.
> 
> Not able to understand what value this brings (maybe I'm wrong re: refactoring? please correct me if that's the case).
> 
> I do see a downside. Having arbitrary keywords will pollute the codebase. I know with `:screen`, your intent is very clear. But there could be a case where navigation opens a `bottom-sheet`. It will be very enticing to have a `:bottom-sheet` keyword in that case. There is no end to this.

We have many qualified keywords like this used for different cases, e.g icons are using `:/i, translations are using `:/t`. it's helps categorise certain aspects of the codebase. 

In the case of translation keywords the intent is obvious because it's basically always using an` i18n/label`.
however, consider for example if we are refactoring and we want to change all uses of the `:password` translation, it is clearly much easier and safer to find the relevant keywords using `:t/password` then a generic search for `:password` and take the context from it's usage - this in turn simplifies the pr review process too.

We have for example some generic screen names here, such as `:shell` - "Shell" is a concept we use all throughout the designs to suggest the overlay pages that are dark theme, regardless of the users theme.
Some other very generic screen names `:settings`, `:new-contact`, `:airdrop-addresses`, `:shell-stack` - there are many more (and I'm sure more will be added), all which can be used in several other contexts within our project. 

Of course, this can go further as shown in your bottom-sheet example, but I think in this case we need specificity of what `:screen/` qualification is used for. In this case, I think it can be simply any screen we have registered in the screens.cljs file. Nothing outside of that. If something else should be used as `:screen/` and is outside the screens.cljs file then it should be moved there imo 👍 
Whether a qualified keyword is then needed for bottom-sheets is another question and so I guess with qualified keywords we can take them on a case by case basis as we are already doing this a lot in the codebase 👌 

That's just my view on this anyway, @ulisesmac also pointed out some good points too 👍

ilmotta commented:
@J-Son89, I didn't read all the comments in this PR, but I am absolutely in favor of qualifying certain keywords to make them unique in the system. I lost count of the amount of times I had to find the `view` definition of a screen, but the keyword had too many hits while grepping (e.g. `:photo-selector`, `settings`, `intro`, and many others).

J-Son89 commented:
> @J-Son89, I didn't read all the comments in this PR, but I am absolutely in favor of qualifying certain keywords to make them unique in the system. I lost count of the amount of times I had to find the `view` definition of a screen, but the keyword had too many hits while grepping (e.g. `:photo-selector`, `settings`, `intro`, and many others).

@ilmotta - I believe it is currently just @shivekkhurana with some concerns here. https://github.com/status-im/status-mobile/pull/18778#issuecomment-1938880350

I have responded - https://github.com/status-im/status-mobile/pull/18778#issuecomment-1948207052

and @ulisesmac  contributed some insights on why he is in preference of this approach - https://github.com/status-im/status-mobile/pull/18778#issuecomment-1939277284

@shivekkhurana have your concerns been somewhat satisfied?

ilmotta commented:
> I think the decision we were basing it off for when we want to use the abbreviated form vs more verbose is how frequently we use these. right @ilmotta? 🤔

Correct @J-Son89, we decided on the short form of `:i` for `:icon` based on frequency, and also because `[quo/icon :icon/foo]` felt too redundant.

I'm not a fan of abbreviating `:screen` to `:s`, but that would work fore sure.

--------

But here's my take on this subject.

These discussions about non-namespaced keyword names are more difficult as well because we are creating _categories_ (qualifications) for things that already have a proper place. With namespaced keywords, we still get keyword auto-complete, jump to source (depending on your editor and settings), and we can alias the namespace as short as we desire (such as `s`). Ah, we also get compile-time protection against typos in the namespace part of a keyword like `::screens/settings`. Namespaced keywords shouldn't always be used though. But if the problem is to uniquely refer to some keyword and differentiate them in the whole system, it's really handy.

It feels a bit like we're skipping this very core feature of Clojure.

One of the reasons I so wanted to upgrade shadow-cljs in PR https://github.com/status-im/status-mobile/pull/15417 was to get the `:as-alias` feature from Clojure 1.11+, which we can use now. `as-alias` eliminates the problem of circular dependencies to refer to keywords.

The code below does not even compile pre Clojure 1.11 if you use `:as` (circular dependency error), but I checked this in our repo with `:as-alias` and it works beautifully.

```clojure
;; Anywhere we want to refer to a screen name:
(require '[status-im.navigation.screens :as-alias screens])

;; Then on that same namespace (note the double colon):
(rf/dispatch [:open-modal ::screens/settings])

;; And in the screens namespace `status-im.navigation.screens` (note the double colon)
{:name      ::settings
 :options   options/transparent-screen-options
 :component settings/view}
```

Clojure gives us a way to differentiate values based on their context. Currently, the keyword `:settings` can be a screen, but also a list type in the quo component `settings.category`. We can refer to different keywords (semantically speaking) by using their namespaces.

J-Son89 committed:
chore: qualify screen keywords

status-im-auto commented:
## 100% of end-end tests have passed
```
Total executed tests: 5
Failed tests: 0
Expected to fail tests: 0
Passed tests: 5
```
<h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3483595">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/111264517fd440dfa002ec143094da3c?auth=886ac6c19cb7262d4d344f51d4c2a525#59">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7896cf0ff2454ddcb286a9eaaadcf22f?auth=c6227b27250a6c087e87dcd5878bc94f#58">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3483597">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bc6d70d0ba16443da55ec129175a9dca?auth=1f1efc2cfa7056244b56a316bba261a9#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3483598">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a08b45c669534fdd86538188652a6f31?auth=c4e439cb247dbe3ea16ded37382db760#132">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e321c7d2e39f43eba57f0e6a15500a1e?auth=aa57f03c31b9a72aad8481ca702f8b50#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3483599">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/30558194ad3d4b4a825b8e87bafafa73?auth=e4eb6a8fc977a4b8d4d54fd9c6e34106#206">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2368b34c552c4f3dac82c915ed50dfd8?auth=cbf75c41b6995440fb3b7f5b7ad10384#165">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3483596">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/30558194ad3d4b4a825b8e87bafafa73?auth=e4eb6a8fc977a4b8d4d54fd9c6e34106#153">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2368b34c552c4f3dac82c915ed50dfd8?auth=cbf75c41b6995440fb3b7f5b7ad10384#109">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

qoqobolo commented:
@J-Son89 thanks for the PR! Take a look at the issue please:

### ISSUE 1: `Cannot set property 'reagent-compiler2' of null` when opening a collectible

**STR:**
1. Open `Collectibles`
2. Press any collectible

https://github.com/status-im/status-mobile/assets/67952253/effba0b7-b558-4152-ae13-3aed39b082e4

qoqobolo commented:
### ISSUE 2: `Enter amount` page issues

There are two issues on this page:
1. Routs/fees are not shown on the screen. But it seems this is something different from https://github.com/status-im/status-mobile/issues/19099, in the video below you can see that this is just a blank page under the amount entry field
2. You cannot go back from this page. When you press the `back` button, the token icon disappears, as well as the `Max` amount below

PR:


https://github.com/status-im/status-mobile/assets/67952253/6c9ce84b-e6e7-4948-8894-46711a2b6df6



Nightly:

https://github.com/status-im/status-mobile/assets/67952253/2b5d8247-f499-481a-a5e0-0011659e2760

**STR:**

1. Go to the `Send` flow
2. Enter a valid amount, wait for routs and fees
3. Tap the `back` button

J-Son89 committed:
chore: adjust to bugs

J-Son89 commented:
@qoqobolo - Apologies about that, I think it was a rebasing issue that they snook in. I addressed all those issues now as they were straightforward problems to fix with the info you gave me 🙏

qoqobolo commented:
> @qoqobolo - Apologies about that, I think it was a rebasing issue that they snook in. I addressed all those issues now as they were straightforward problems to fix with the info you gave me 🙏

No worries, the main thing is that we didn’t let them sneak into develop! 🙂
And thanks for the fix! Could you rebase the branch and resolve the conflict, please? I'll re-check it

J-Son89 committed:
Merge branch 'develop' into jc/refactor-screens
